### ISSUE TITLE
Add tests and sample code for constructing an OpsLogger with Dagger

### ISSUE THREAD

**Issue by seanjreilly** [EqualExperts/opslogger#11]
[Dagger](http://square.github.io/dagger/) is a new dependency injection system, and it's used by at least one member of the OpsLogger community.

Add a unit test to OpsLoggerFactory to ensure that creating OpsLogger instances in Dagger works, and add some code to `sample-usage` to illustrate how this is done.

For a typical case, this might look something like this:

```java
@Provides OpsLogger<Foo> logger() {
    return new OpsLoggerFactory()
        .setDestination(System.err)
        //customise here
        .build()
}
```

@ravn, your use case is more complicated, but it looks like this might work:

```java
@Provides OpsLoggerFactory opsLoggerFactory() {
    return new OpsLoggerFactory()
        .setDestination(System.err);
        //customise here
}

@Provides OpsLoggerFactory<Foo> fooLogger(OpsLoggerFactory factory) {
    return factory.build();
}

@Provides OpsLoggerFactory<Bar> barLogger(OpsLoggerFactory factory) {
    return factory.build();
}
```

**Comment by seanjreilly** [comment]
Partial progress to report on this front:

I've pushed 935de00, which adds a simple dagger test. The annotation processing tool integrated very
smoothly with gradle, and I got everything to work in the IDE just by turning on auto-discovery of annotation processors from the classpath, which seems reasonable.

The only remaining issue is that dagger complains that `build()` throws a checked IOException. With some refactoring, build can be made to throw an UncheckedIOException instead.
